release/ray_release/test_automation/state_machine.py

```diff
@@ -28,20 +39,44 @@ def __init__(self, test: Test) -> None:
             TestState.JAILED: TestState.PASSING
             if self._jailed_to_passing()
             else TestState.JAILED,
-            # passing -> failing
+            # passing -> failing OR passing -> consistently_failing
```

Comment:
this state transition map is becoming much harder to parse ..

maybe change this into a lambda/function ? like `def _next_state(self, current_state):`, and just use simple `if/else` statements?